Jint.Benchmark/UncacheableExpressionsBenchmark.cs

```diff
@@ -0,0 +1,192 @@
+
+namespace Jint.Benchmark
+{
+    /// <summary>
+    /// Test case for situation where object is projected via filter and map, Jint deems code as uncacheable.
+    /// </summary>
+    [Config(typeof(Config))]
+    public class UncacheableExpressionsBenchmark
+    {
+        private class Config : ManualConfig
+        {
+            public Config()
+            {
+                Add(Job.MediumRun);
+                Add(MemoryDiagnoser.Default);
+            }
+        }
+
+        private Document doc;
+
+        private string targetObject;
+        private JsValue[] targetJsObject;
+
+        private const string script = @"
+function output(d) {
+    var doc = d.SubDocuments.find(function(x){return x.Id==='testing';});
+    return { Id : d.Id, Deleted : d.Deleted, SubTestId : (doc!==null&&doc!==undefined)?doc.Id:null, Values : d.SubDocuments.map(function(x){return {TargetId:x.TargetId,TargetValue:x.TargetValue,SubDocuments:x.SubDocuments.filter(function(s){return (s!==null&&s!==undefined);}).map(function(s){return {TargetId:s.TargetId,TargetValue:s.TargetValue};})};}) };
+}
+";
+
+        private Engine engine;
+
+        public class Document
+        {
+            public string Id { get; set; }
+            public string TargetId { get; set; }
+            public decimal TargetValue { get; set; }
+            public bool Deleted { get; set; }
+            public IEnumerable<Document> SubDocuments { get; set; }
+        }
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            doc = new Document
+            {
+                Deleted = false,
+                SubDocuments = new List<Document>
+                {
+                    new Document
+                    {
+                        TargetId = "id1",
+                        SubDocuments = Enumerable.Range(1, 200).Select(x => new Document()).ToList()
+                    },
+                    new Document
+                    {
+                        TargetId = "id2",
+                        SubDocuments = Enumerable.Range(1, 200).Select(x => new Document()).ToList()
+                    }
+                }
+            };
+
+            using (var stream = new MemoryStream())
+            {
+                using (var writer = new StreamWriter(stream))
+                {
+                    JsonSerializer.CreateDefault().Serialize(writer, doc);
+                    writer.Flush();
+
+                    var targetObjectJson = Encoding.UTF8.GetString(stream.ToArray());
+                    targetObject = $"var d = {targetObjectJson};";
+                }
+            }
+
+            CreateEngine();
+        }
+
+        private static void InitializeEngine(Options options)
+        {
+            options
+                .LimitRecursion(64)
+                .MaxStatements(int.MaxValue)
+                .Strict()
+                .LocalTimeZone(TimeZoneInfo.Utc);
+        }
+
+        [Params(500)]
+        public int N { get; set; }
```

Comment:
I don't understand this. Why is it necessary when Benchmarks.NET can already do that by itself?